### PR TITLE
refactor: add config_signature field inside operator in definition

### PR DESCRIPTION
**Pull request by xenowits**
1. Add `config_signature` field inside `Operator` struct in cluster definition.
2. Remove `operator_signatures` field
3. More details here: https://github.com/ObolNetwork/charon/issues/646#issuecomment-1143750714

category: refactor
ticket: #646

### CODE REVIEW

File: cluster/confighash.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import ssz "github.com/ferranbt/fastssz"
+
+// staticDefinition defines the static (non-changing) portion of the charon cluster definition.
+type staticDefinition struct {
```
Comment (xenowits) [inline cluster/confighash.go:21]: needed to create this as this seems the only easy way to derive ssz hashes on. I wanted to extract the common fields of the definition object, but i hope a little duplication is okay?

Comment (dB2510) [reply cluster/confighash.go:21]: yeah, the fields of this struct can be private though

File: cluster/operator.go [hunk 2]
```diff
@@ -25,15 +25,18 @@ import (
...
 type Operator struct {
-	// Address is the Ethereum address identifying the operator.
+	// The Ethereum address of the operator
 	Address string `json:"address"`
 
 	// ENR identifies the charon node.
 	ENR string `json:"enr"`
 
-	// Nonce is incremented each time the ENR is signed.
+	// Nonce is incremented each time the ENR or the ENRSignature is changed
```
Comment (dB2510) [inline cluster/operator.go:34]:
```suggestion
	// Nonce is incremented each time the ENR is added/signed
```
maybe

File: cluster/confighash.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	}
+
+	hh.Merkleize(indx)
+
+	return nil
+}
+
+// ConfigHash returns the config hash of the given cluster definition object. The config hash is the
+// ssz hash of all the static fields of the definition object and hence doesn't change once created.
+func ConfigHash(d Definition) ([32]byte, error) {
```
Comment (dB2510) [inline cluster/confighash.go:107]: this can be a private function, since we have a public method on definition

File: cluster/definition.go [hunk 4]
```diff
@@ -125,46 +121,42 @@ func (d Definition) NodeIdx(pID peer.ID) (NodeIdx, error) {
...
-				return false, err
-			} else if ok {
-				found = true
-				break
-			}
-		}
-
-		if !found {
-			return false, nil
+		if ok, err := verifySig(o.Address, digest[:], o.ConfigSignature); err != nil {
```
Comment (dB2510) [inline cluster/definition.go:138]: likewise verification of ENR signatures can also be added

**Review by dB2510 — APPROVED** [review]
LGTM!
docs/configuration.md and docs repo needs to be updated as well

File: cluster/definition.go (hunk 4 above)
Comment (xenowits) [reply cluster/definition.go:138]: yeah

**Comment by xenowits** [comment]
> LGTM! docs/configuration.md and docs repo needs to be updated as well

I have updated `docs/configuration.md` but need to update the docs repo later on

**Pull request merged**